src/rpc/chain_api.rs

```diff
@@ -241,6 +243,69 @@ pub async fn chain_get_block_messages<DB: Blockstore>(
     Ok(ret)
 }
 
+/// Find the path between two tipsets, as a series of [`PathChange`]s.
+///
+/// ```text
+/// 0 - A - B - C - D
+///     ^~~~~~~~> apply B, C
+///
+/// 0 - A - B - C - D
+///     <~~~~~~~^ revert C, B
+///
+///     <~~~~~~~~ revert C, B
+/// 0 - A - B  - C
+///     |
+///      -- B' - C'
+///      ~~~~~~~~> then apply B', C'
+/// ```
+///
+/// Exposes errors from the [`Blockstore`], and returns an error if there is no common ancestor.
+pub async fn chain_get_path<DB: Blockstore>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((from, to))): Params<LotusJson<(TipsetKey, TipsetKey)>>,
+) -> Result<LotusJson<Vec<PathChange>>, JsonRpcError> {
+    impl_chain_get_path(&data.chain_store, &from, &to)
+        .map(LotusJson)
+        .map_err(Into::into)
+}
+
+fn impl_chain_get_path(
+    chain_store: &ChainStore<impl Blockstore>,
+    from: &TipsetKey,
+    to: &TipsetKey,
+) -> anyhow::Result<Vec<PathChange>> {
+    let mut to_revert = chain_store
+        .load_required_tipset(from)
+        .context("couldn't load `from`")?;
+    let mut to_apply = chain_store
+        .load_required_tipset(to)
+        .context("couldn't load `to`")?;
+
+    let mut all_reverts = vec![];
+    let mut all_applies = vec![];
+
+    while to_revert != to_apply {
```

Comment:
Are we sure it won't go wild into an infinite loop? I trust your judgement. It's not apparent at first glance.